src/services/RoomService.test.ts

```diff
@@ -0,0 +1,60 @@
+import sinon from "sinon";
+import { stubInterface } from "ts-sinon";
+import Room  from "../entities/Room";
+import getRoomMock from "../mocks/RoomMock";
+import IRepository from "../types/IRepository";
+import RoomService from "./RoomService";
+
+
+describe(
+  "RoomService", () => {
+    const sandbox = sinon.createSandbox();
+
+    afterEach(() => sandbox.restore());
+
+    it(
+      "create success", async () => {
+        const roomMock = getRoomMock();
+        const fakeRepo = stubInterface<IRepository<Room>>();
+        const userInput = {
+          name: roomMock.name,
+        };
+
+        // When
+        fakeRepo.save.resolves(roomMock);
+        sandbox.replace(
+          RoomService.prototype, "getRepository", () => fakeRepo,
+        );
+
+        // Then
+        const res = await new RoomService().create(userInput);
+
+        expect(fakeRepo.save.calledOnceWithExactly(userInput)).toBeTruthy();
+        expect(res).toEqual(roomMock);
+
+      },
+
+    );
+
+    it(
+      "create fails due to missing params", async () => {
```

Comment:
We should assume that the RoomService will always receive correct parameters and that exception would be thrown by validator middleware.

Furthermore, you are testing something that this service does not control, you are assuming that the fakeRepo will throw if no name is provided, that sounds like an appropriate test for the repository but not for the service. If we provide an implementation of a repository that does not throw, then service won't throw neither.

I'd just remove this test.